lib/internal/test_runner/mock.js

```diff
@@ -157,18 +161,40 @@ class MockTracker {
         'options.setter', setter, "cannot be used with 'options.getter'"
       );
     }
+    function getOriginalObject(descriptor) {
+      let original;
+
+      if (getter) {
+        original = descriptor?.get;
+      } else if (setter) {
+        original = descriptor?.set;
+      } else {
+        original = descriptor?.value;
+      }
+
+      return original;
+    }
 
-    const descriptor = ObjectGetOwnPropertyDescriptor(object, methodName);
-    let original;
-
-    if (getter) {
-      original = descriptor?.get;
-    } else if (setter) {
-      original = descriptor?.set;
-    } else {
-      original = descriptor?.value;
+    function findMethodOnPrototypeChain(instance, method, isDescriptor = false) {
```

Comment:
This function seems more complicated than it needs to be. Would it work to do something like this to get the host object and the descriptor (untested):

```js
let host = object;
let descriptor;

while (host !== null) {
  descriptor = ObjectGetOwnPropertyDescriptor(host, methodName);

  if (descriptor) {
    break;
  }

  host = ObjectGetPrototypeOf(host);
}
```

Reply:
loved it! just tested and commited